### PR TITLE
Validation refactor

### PR DESCRIPTION
**Pull request by 0505gonzalez**
Problem:
Validation does not apply Luhn algorithm

Solution:
* Create new CardValidator class
* Implement Luhn algorithm
* Add unit test coverage for CardValidator
* Deprecate Xendit validation methods

### CODE REVIEW

File: xendit-android/src/main/java/com/xendit/utils/CardValidator.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        if (str == null) {
+            return null;
+        }
+
+        return str.replaceAll("\\s", "");
+    }
+
+    private static boolean isNumeric(String str) {
+        try {
+            Double.parseDouble(str);
```
Comment (afik) [inline xendit-android/src/main/java/com/xendit/utils/CardValidator.java]: I'm not sure but can Double hold 19 digit card number?

File: app/src/main/java/com/xendit/example/AuthenticationActivity.java [hunk 2]
```diff
@@ -15,11 +15,11 @@
 
 import com.xendit.AuthenticationCallback;
 import com.xendit.Models.Authentication;
-import com.xendit.Models.Token;
 import com.xendit.Models.XenditError;
-import com.xendit.TokenCallback;
 import com.xendit.Xendit;
 
+import ch.itomy.xendit_example.R;
```
Comment (afik) [inline app/src/main/java/com/xendit/example/AuthenticationActivity.java]: why this isn't change to com.xendit too?

File: xendit-android/src/main/java/com/xendit/utils/CardValidator.java (hunk 1 above)
Comment (0505gonzalez) [inline xendit-android/src/main/java/com/xendit/utils/CardValidator.java]: Very good catch

File: app/src/main/java/com/xendit/example/AuthenticationActivity.java (hunk 2 above)
Comment (0505gonzalez) [inline app/src/main/java/com/xendit/example/AuthenticationActivity.java]: Fixed

**Comment by afik** [comment]
LGTM

**Pull request merged**